pynisher/util.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+from typing import Any, Callable
+
+import psutil
+from psutil import Process
+
+_mem_unit_table = {
+    "B": 1,
+    "KB": 2**10,
+    "MB": 2**20,
+    "GB": 2**30,
+}
+
+_time_unit_table = {"s": 1, "m": 60, "h": 60 * 60}
+
+
+def memconvert(x: float, *, frm: str = "B", to: str = "B") -> float:
+    """Convert between memory units
+
+    Parameters
+    ----------
+    x : float
+        The memory amount
+
+    frm : "B" | "KB" | "MB" | "GB" = "B"
+        What unit it is in
+
+    to :  "B" | "KB" | "MB" | "GB" = "B"
+        What unit to convert to
+
+    Returns
+    -------
+    float
+        The memory amount
+    """
+    u_from = _mem_unit_table[frm.upper()]
+    u_to = _mem_unit_table[to.upper()]
+
+    as_bytes = x * u_from
+    as_target = as_bytes / u_to
+
+    # We can't see a use case for float Bytes
+    if to.upper() == "B":
+        return int(as_target)
```

Comment:
Can there be rounding errors in the computation above? If yes, would it be better to use `rint` here?

Reply:
Ahh nvm it's a `np.rint`, but we can use the built in `round` in python.